components/add_users_to_group_modal/add_users_to_group_modal.tsx

```diff
@@ -0,0 +1,154 @@
+    };
+
+    const addUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const deleteUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const addUsersToGroup = async (users?: UserProfile[]) => {
+        if (!users || users.length === 0) {
+            return;
+        }
+        const userIds = users.map((user) => {
+            return user.id;
+        });
+
+        const data = await props.actions.addUsersToGroup(props.groupId, userIds);
+
+        if (data?.error) {
+            setShowUnknownError(true);
+        } else {
+            goBack();
+        }
+    };
+
+    const goBack = () => {
+        props.backButtonCallback();
+        props.onExited();
+    };
+
+    return (
+        <Modal
+            dialogClassName='a11y__modal user-groups-modal-create'
+            show={show}
+            onHide={doHide}
+            onExited={props.onExited}
+            role='dialog'
+            aria-labelledby='createUserGroupsModalLabel'
+            id='addUsersToGroupsModal'
+        >
+            <Modal.Header closeButton={true}>
+                <button
+                    type='button'
+                    className='modal-header-back-button btn-icon'
+                    aria-label='Close'
+                    onClick={() => {
+                        goBack();
+                    }}
+                >
+                    <LocalizedIcon
+                        className='icon icon-arrow-left'
+                        ariaLabel={{id: t('user_groups_modal.goBackLabel'), defaultMessage: 'Back'}}
+                    />
+                </button>
+                <Modal.Title
+                    componentClass='h1'
+                    id='addUsersToGroupsModalLabel'
+                >
+                    <FormattedMessage
+                        id='user_groups_modal.addPeopleTitle'
+                        defaultMessage='Add people to {group}'
+                        values={{
+                            group: props.group.display_name,
+                        }}
+                    />
+                </Modal.Title>
+            </Modal.Header>
+            <Modal.Body
+                className='overflow--visible'
+            >
+                <div className='user-groups-modal__content'>
+                    <form role='form'>
+                        <div className='group-add-user'>
+                            <AddUserToGroupMultiSelect
+                                multilSelectKey={'addUsersToGroupKey'}
+                                onSubmitCallback={addUsersToGroup}
+                                focusOnLoad={false}
+                                savingEnabled={isSaveEnabled()}
+                                addUserCallback={addUserCallback}
+                                deleteUserCallback={deleteUserCallback}
+                                groupId={props.groupId}
+                                searchOptions={{
```

Comment:
This object is going to change in every render, probably we can generate a make it stable with a `useMemo`